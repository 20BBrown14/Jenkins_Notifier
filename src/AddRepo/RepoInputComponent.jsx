import React from 'react';
import InputField from 'terra-form-input/lib/InputField';
import PropTypes from 'prop-types';


const propTypes = {
/** Error message */
  errorMessage: PropTypes.string.isRequired,
  /** Whether form input is invalid */
  formInvalid: PropTypes.bool,
  /** Function for when repo url input field changes */
  URLFieldChange: PropTypes.func.isRequired,
  /** Function for when repo name input field changes */
  nameFieldChanged: PropTypes.func.isRequired,
  /** Help message to display under input field */
  helpMessage: PropTypes.string,
  /** Value of the repo URL input field */
  repoURL: PropTypes.string.isRequired,
  /** Value of the repo name input field */
  repoName: PropTypes.string.isRequired,
  /** Whether or not the view is loading */
  isLoading: PropTypes.bool,
};

const defaultProps = {
  formInvalid: false,
};

const RepoInputComponent = (props) => {
  const {
    errorMessage,
    formInvalid,
    URLFieldChange,
    helpMessage,
    repoURL,
    repoName,
    nameFieldChanged,
    isLoading,
  } = props;
  return (
    <div>
      <InputField
        style={{ margin: '15px 25px 12.5px 25px' }}
        inputId="repoNameInput"
        label="Name"
        value={repoName}
        onChange={nameFieldChanged}
        help="Input no name to use name on Jenkins"
        inputAttrs={{ disabled: isLoading }}
      />
      <InputField
        style={{ margin: '12.5px 25px 12.5px 25px' }}
        inputId="repoURLInput"
        label="URL"
        error={errorMessage}
        help={helpMessage}
        isInvalid={formInvalid}
        onChange={URLFieldChange}
        value={repoURL}
        inputAttrs={{ disabled: isLoading }}
      />
    </div>
  );
};

RepoInputComponent.propTypes = propTypes;
RepoInputComponent.defaultProps = defaultProps;

export default RepoInputComponent;
