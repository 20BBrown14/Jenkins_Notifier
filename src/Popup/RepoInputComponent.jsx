import React from 'react';
import InputField from 'terra-form-input/lib/InputField';
import PropTypes from 'prop-types';


const propTypes = {
/** Error message */
  errorMessage: PropTypes.string.isRequired,
  /** Whether form input is invalid */
  formInvalid: PropTypes.bool,
  /** Function for when repo url input field changes */
  URLFieldChange: PropTypes.func.isRequired,
  /** Function for when repo name input field changes */
  nameFieldChanged: PropTypes.func.isRequired,
  /** Help message to display under input field */
  helpMessage: PropTypes.string,
  /** Value of the repo URL input field */
  repoURL: PropTypes.string.isRequired,
  /** Value of the repo name input field */
  repoName: PropTypes.string.isRequired,
};

const defaultProps = {
  formInvalid: false,
};

const RepoInputComponent = (props) => {
  const {
    errorMessage,
    formInvalid,
    URLFieldChange,
    helpMessage,
    repoURL,
    repoName,
    nameFieldChanged,
  } = props;

  return (
    <div>
      <InputField
        style={{ margin: '15px 25px 12.5px 25px' }}
        inputId="repoNameInput"
        label="Name"
        value={repoName}
        onChange={nameFieldChanged}
      />
      <InputField
        style={{ margin: '12.5px 25px 12.5px 25px' }}
        inputId="repoURLInput"
        label="URL"
        error={errorMessage}
        help={helpMessage}
        isInvalid={formInvalid}
        onChange={URLFieldChange}
        value={repoURL}
      />
    </div>
  );
};

RepoInputComponent.propTypes = propTypes;
RepoInputComponent.defaultProps = defaultProps;

export default RepoInputComponent;
