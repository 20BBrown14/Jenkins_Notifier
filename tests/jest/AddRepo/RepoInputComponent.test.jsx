import React from 'react';
import RepoInputComponent from '../../../src/AddRepo/RepoInputComponent';

describe('RepoInputComponent', () => {
  it('should render a default view', () => {
    const testComponent = (shallow(
      <RepoInputComponent
        errorMessage="someMessage"
        handleValidation={() => {}}
        repoURL=""
        repoName=""
        URLFieldChange={() => {}}
        nameFieldChanged={() => {}}
        confirmClickHandler={() => {}}
      />,
    ));
    expect(testComponent).toMatchSnapshot();
  });
});
