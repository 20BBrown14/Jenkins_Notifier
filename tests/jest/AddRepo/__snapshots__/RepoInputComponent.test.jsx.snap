// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoInputComponent should render a default view 1`] = `
<div>
  <InputField
    error={null}
    errorIcon={
      <IconError
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    }
    help="Input no name to use name on Jenkins"
    hideRequired={false}
    inputAttrs={
      Object {
        "disabled": undefined,
        "onKeyPress": [Function],
      }
    }
    inputId="repoNameInput"
    isInline={false}
    isInvalid={false}
    isLabelHidden={false}
    label="Name"
    labelAttrs={Object {}}
    onChange={[Function]}
    required={false}
    showOptional={false}
    style={
      Object {
        "margin": "15px 25px 12.5px 25px",
      }
    }
    value=""
  />
  <InputField
    error="someMessage"
    errorIcon={
      <IconError
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    }
    help={null}
    hideRequired={false}
    inputAttrs={
      Object {
        "disabled": undefined,
        "onKeyPress": [Function],
      }
    }
    inputId="repoURLInput"
    isInline={false}
    isInvalid={false}
    isLabelHidden={false}
    label="URL"
    labelAttrs={Object {}}
    onChange={[Function]}
    required={false}
    showOptional={false}
    style={
      Object {
        "margin": "12.5px 25px 12.5px 25px",
      }
    }
    value=""
  />
</div>
`;
